tests/50federation/33room-get-missing-events.pl

```diff
@@ -377,21 +377,49 @@ sub sytest_user_and_room_fixture {
             Future->done(1);
          }),
       )->then( sub {
-         # wait for S to turn up
-         await_sync_timeline_contains(
-            $creator_user, $room2->room_id, check => sub {
-               my ( $event ) = @_;
-               log_if_fail "Got event in room2", $event;
+         # the server may send a /state request; be prepared to answer that.
+         # (it may, alternatively, send individual /event requests)
+         my $state_req_fut = $inbound_server->await_request_v1_state(
+            $room2->{room_id}, $event_id_Q,
+         )->then( sub {
```

Comment:
hrm, apparently it is - which means that it's going to ignore the event_id, and *any* `/state` will satisfy it, rather than one specifically for this event. 

Fixed in ca8057b9.